ext/mysqli/tests/mysqli_character_set.phpt

```diff
@@ -44,7 +44,7 @@ if (!function_exists('mysqli_set_charset')) {
         /* The server currently 17.07.2007 can't handle data sent in ucs2 */
         /* The server currently 16.08.2010 can't handle data sent in utf16 and utf32 */
         /* The server currently 02.09.2011 can't handle data sent in utf16le */
-        if ($charset['Charset'] == 'ucs2' || $charset['Charset'] == 'utf16' || $charset['Charset'] == 'utf32' || 'utf16le' == $charset['Charset']) {
+        if ($charset['Charset'] == 'ucs2' || $charset['Charset'] == 'utf16' || $charset['Charset'] == 'utf32' || 'utf16le' == $charset['Charset'] || 'utf8mb3' == $charset['Charset']) {
```

Comment:
Shouldn't that say `utf8` instead? Also, please add a comment on why we are doing that.

Reply:
The problem is that `utf8mb3` is reported by `SHOW CHARACTER SET`, but no longer allowed to be used as character set. Looks like upstream screwed that up, but somehow we need to cater to that.

I've added respective comments.

Reply:
Ok, I am investigating this now and here is why I found out so far. 

MySQL 8.0.29 changed the output of [`SHOW CHARACTER SET`](https://dev.mysql.com/doc/relnotes/mysql/8.0/en/news-8-0-29.html). This change is consistent with changes they have been making since at least 8.0.26 to the output to slowly phase out `utf8` as an alias of `utf8mb3`. It future, they have plans to swap `utf8` to be an alias of `utf8mb4`. 

Here's the problem now. Our mysqlnd doesn't support `utf8mb3` charset... This would mean that the test failures are valid. 

According to [MySQL docs](https://dev.mysql.com/doc/refman/8.0/en/charset-connection.html#charset-connection-impermissible-client-charset) the only values we should skip in the tests are:

- ucs2
- utf16
- utf16le
- utf32

There's no mention of `utf8mb3` being invalid connection character set. There's also no mention of `utf8` being invalid either. I am not sure what to do with this information. We should probably patch mysqlnd up to allow for these charsets. They have no future, but until these charsets are phased out and companies upgrade to this MySQL version might take 10 or 20 years. If clients are allowed to use `utf8mb3` connection charset now, we should allow it in PHP.

Reply:
> Here's the problem now. Our mysqlnd doesn't support utf8mb3 charset...

Ah, so this is a mysqlnd problem only!

> We should probably patch mysqlnd up to allow for these charsets.

Yeah, that likely makes the most sense.

Anyhow, since this patch is not a proper solution, I'm closing the PR.